feature/autodiscovery/api/src/main/kotlin/app/k9mail/autodiscovery/api/AutoDiscoveryResult.kt

```diff
@@ -51,11 +52,11 @@ sealed interface AutoDiscoveryResult {
  *
  * Implementations contain protocol-specific properties.
  */
-interface IncomingServerSettings
+interface IncomingServerSettings : HasProtocolType
```

Comment:
We already have specific implementations of this interface for each protocol. Also having a `protocolType` property feels redundant. 

`ProtocolType` is currently not used in any meaningful way in this PR. Maybe leave that change to a PR that actually makes use of this new functionality?

Reply:
Right, it's actually for a different problem that should be on another pull-request.